common/src/main/java/com/microsoft/hydralab/common/screen/ADBScreenRecorder.java

```diff
@@ -93,12 +93,14 @@ public void startRecord(int maxTimeInSecond) {
                         recordingProcess.destroy();
                     }
                     deviceInfo.finishCommand();
+                    // make sure the recording procedure is stopped completely
+                    ThreadUtil.safeSleep(2000);
 
                     String outputFilePrefix = new File(baseFolder, DateUtil.fileNameDateDashFormat.format(new Date())).getAbsolutePath();
-
                     final String outFileFullPath = outputFilePrefix + "_" + totalTime + "_" + (totalTime + timeSpan) + ".mp4";
                     String pullComm = String.format("pull %s %s", fileName, outFileFullPath);
                     Process process = adbOperateUtil.executeDeviceCommandOnPC(deviceInfo, pullComm, logger);
```

Comment:
please verify the file size after pulling

Reply:
Updated with retry logic, thx.